### PR TITLE
Utku / Class 19 / Homework Week 2

### PR DESCRIPTION
**Pull request by u61**

### CODE REVIEW

File: week2/homework/src/index.js [hunk 1]
```diff
@@ -1,3 +1,41 @@
...
+const help = require('./intro.js');
+const add = require('./add.js');
+const update = require('./update.js');
+const list = require('./list.js');
+const remove = require('./remove.js');
+const reset = require('./reset.js');
+
+/** ---------------------------------------
+     Commands
+ ---------------------------------------- */
```
Comment (saragerion) [inline week2/homework/src/index.js]: Great job with the commander here! Don't forget to handle the case when no command is specified (should display the help function)

File: week2/homework/src/reset.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ ---------------------------------------- */
+const reset = {
+  execute: () => {
+    const content = fs.readFileSync('To-Do Database.json', 'utf8');
+    switch (content) {
+      case '':
+        console.log('There is nothing to remove !');
+        break;
+      default:
+        fs.writeFile('To-Do Database.json', '', err => {
```
Comment (saragerion) [inline week2/homework/src/reset.js]: In alternative of `''`, you could write an empty array: `'[]'`.
Doing that, you don't need to use fillEmptyDatabase in the add functionality

File: week2/homework/src/add.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+     Base
+ ---------------------------------------- */
+const fs = require('fs');
+
+/** ---------------------------------------
+     Functions for Command
+ ---------------------------------------- */
+function fillEmptyDatabase() {
+  const content = fs.readFileSync('To-Do Database.json', 'utf8');
+  switch (content) {
```
Comment (saragerion) [inline week2/homework/src/add.js]: Switch is used when you have multiple scenario's (condition). If there is only 1 condition it makes more sense to use an `if` statement

File: week2/homework/src/update.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    switch (content) {
+      case '':
+        console.log('There is nothing to update !');
+        break;
+      default:
+        fs.readFile('To-Do Database.json', 'utf8', (err, readData) => {
+          if (err) throw err;
+          const parsedJSON = JSON.parse(readData);
+          const arrayLength = parsedJSON.length;
+          if (toDoNumber < 0 || toDoNumber > arrayLength - 1) {
```
Comment (saragerion) [inline week2/homework/src/update.js]: Watch out about this bit: `arrayLength - 1`.
What happens if I have only 1 item in the list, and I try to do:
node index.js update 1 "Random string"

File: week2/homework/src/index.js (hunk 1 above)
Comment (u61) [reply week2/homework/src/index.js]: We could use 'node index.js' or 'node index.js help' for showing help section as per assignment. That's why I preffered index.js help. But, I've added commandless option as well.

File: week2/homework/src/reset.js (hunk 2 above)
Comment (u61) [reply week2/homework/src/reset.js]: But the client doesn't call 'reset' before each command. Also, the client always start with an empty database. Therefore, 'add' command always check for if it is an empty database.

File: week2/homework/src/add.js (hunk 3 above)
Comment (u61) [reply week2/homework/src/add.js]: It's good to learn that. All switches have been switched to if statement.

File: week2/homework/src/update.js (hunk 4 above)
Comment (u61) [reply week2/homework/src/update.js]: Yes. It's been corrected.

**Comment by u61** [comment]
Hello @saragerion ,
I've tried to correct my code as per your comments.
Thank you for your review.

**Pull request closed without merging**